### PR TITLE
Loadout Builder fixes

### PR DESCRIPTION
**Pull request by kyleshay**
Working on some loadout builder fixes.

This fixes:

#582 (wrong items being added to loadout)
#585 (always show locked section)

### CODE REVIEW

File: app/scripts/minmax/dimMinMax.controller.js [hunk 1]
```diff
@@ -385,6 +361,7 @@
             return item.primStat &&
               item.primStat.statHash === 3897883278 && // has defence hash
               ((vm.showBlues && item.tier === 'Rare') || item.tier === 'Legendary' || (vm.showExotics && item.tier === 'Exotic')) &&
+              item.primStat.value >= 280 // only 280+ light items
```
Comment (kyleshay) [inline app/scripts/minmax/dimMinMax.controller.js]: i added this in also. only look at items >= 280 defense. reasoning is we only can predict items at 335 if they are >= 280.

**Comment by ericnelson0** [comment]
These are really good suggestions and would be really good changes. Looking forward to testing these out.

**Comment by ericnelson0** [comment]
Could I recommend adding an attribute: <code>lockedChanged: false</code> that is set to true in the event handlers as:

```
      onDrop: function(dropped_id, type) {
          dropped_id = dropped_id.split('-')[1];
          if(vm.lockeditems[type] && vm.lockeditems[type].id == dropped_id) { return; }
          var item = _.findWhere(buckets[vm.active][type], {id: dropped_id});
          vm.lockeditems[type] = item;
          var bestarmor = getBestArmor(buckets[vm.active], vm.lockeditems);
          vm.highestsets = vm.getSetBucketsStep(vm.active, bestarmor);
          if(vm.progress < 1.0) {
            vm.lockedchanged = true;
          }
      },
```
and
```
      onRemove: function(removed_type) {
          vm.lockeditems[removed_type] = null;
          var bestarmor = getBestArmor(buckets[vm.active], vm.lockeditems);
          vm.highestsets = vm.getSetBucketsStep(vm.active, bestarmor);
          if(vm.progress < 1.0) {
            vm.lockedchanged = true;
          }
      },
```

So that the processing can be cancelled in the <code>step</code> function as:
```
                    if((processed_count%5000) == 0) {
                        // If active gaurdian or page is changed then stop processing combinations
                        if(vm.active !== activeGaurdian || vm.lockedchanged == true || $location.path() !== '/best') {
                            if(vm.lockedchanged == true) { vm.lockedchanged  = false; }
                            return;
                        }
                        vm.progress = processed_count/combos;
                        $timeout(step, 0, true, activeGaurdian, h,g,c,l,ci,gh,ar,processed_count);
                        return;
                    }
```
and the necessary:
```
                // Finish progress
                vm.progress = processed_count/combos;
                vm.lockedchanged = false;
                console.timeEnd('elapsed');
```

This will cancel processing once the locked items are changed.

**Comment by kyleshay** [comment]
that's a good call, i'll make sure thats in the next PR i make. going to try and track down that NaN issue tomorrow.

**Pull request merged**